onnxruntime/contrib_ops/cuda/bert/attention_impl.cu

```diff
@@ -762,12 +762,8 @@ Status UnfusedAttention(
   } else {  // no mask
     if (nullptr != data.output_qk) {
       int64_t qk_size = (int64_t)batch_size * num_heads * sequence_length * total_sequence_length;
-      if (std::is_same<T, QK>::value) {
```

Comment:
I guess that adding a constexpr in this line  without the remaining change might also fix the issue.